src/NuGet.Clients/NuGet.PackageManagement.UI/PackageLoadContext.cs

```diff
@@ -50,6 +53,80 @@ internal class PackageLoadContext
 
         public Task<PackageCollection> GetInstalledPackagesAsync() =>_installedPackagesTask;
 
+        // get lock file for projects
+        public async Task<List<LockFile>> getLockFiles(IEnumerable<NuGetProject> nuGetProjects)
+        {
+            var nuGetProjectList = nuGetProjects.ToList();
+            var buildIntegratedProjects = nuGetProjectList.OfType<BuildIntegratedNuGetProject>().ToList();
+
+            List<LockFile> lockFiles = new List<LockFile>();
+            var lockFileFormat = new LockFileFormat();
+
+            foreach (var project in buildIntegratedProjects)
```

Comment:
We probably need an abstraction on NuGetProject for this. 

We should also consider paralleling this work.

Reply:
I also could be wrong, but trying to roughly follow the code path this is happening on the UI thread, and will cause UI delays.

Reply:
I've removed the code that gets dependency packages for PR-style projects from the assets file. We agreed that for right now we would only make recommendations for PC-style projects while we find a better way to get the dependency packages for PR-style projects.

Reply:
GetSupportedFrameworks is the method currently doing it for PC, so we can make improve that.

Reply:
I removed the added GetTargetFrameworksAsync method since it was only added to support PR-style projects. I agree it would be better to improve the existing GetSupportedFrameworks method to add support for PR-style projects.